### ISSUE TITLE
sim: terminal is garbled when nuttx is killed

### ISSUE THREAD

**Issue by v01d** [apache/incubator-nuttx#2163]
If you kill nuttx sim binary (even with SIGTERM), the terminal is garbled. Probably something needs to be done in some exit/signal handler. 

![image](https://user-images.githubusercontent.com/161706/97634419-6de00c80-1a14-11eb-800d-4da879d266ba.png)

**Comment by xiaoxiang781216** [comment]
This is because sim use atexit to restore the state:
https://github.com/apache/incubator-nuttx/blob/master/arch/sim/src/sim/up_simuart.c#L112
https://github.com/apache/incubator-nuttx/blob/master/arch/sim/src/sim/up_x11framebuffer.c#L262
which need change to signal handler, but I guess x11 can restore automatically without the special action.

**Comment by xiaoxiang781216** [comment]
Should be fixed by: https://github.com/apache/incubator-nuttx/pull/2998. @v01d please try it.

**Comment by v01d** [comment]
It would seem so, but I'm confused as to why that PR fixes it. As you mentioned before, there was cleanup required to be done on sim exit.

**Comment by xiaoxiang781216** [comment]
> It would seem so, but I'm confused as to why that PR fixes it. As you mentioned before, there was cleanup required to be done on sim exit.

Since the PR doesn't disable OPOST, \n will always translate to \r\n.

**Comment by v01d** [comment]
Ah, I understand: it was garbled due to changing this bit from default. I assume other changes do not require any undoing?

**Comment by v01d** [comment]
I think that this needs further work since I find now that CTRL+C does not work anymore when nuttx is killed. The cleanest would probably be to get the flags before sim starts and then reapply them in the end.

**Comment by xiaoxiang781216** [comment]
Yes, the input process is turned off, we need to find a reliable way to restore it.  #2998 just fix the output:(.

**Comment by v01d** [comment]
I'll reopen for tracking
